### PR TITLE
DTIN-3938: Accommodate busy k8s clusters with short-lived pods

### PR DESCRIPTION
**Pull request by jmakar-s1**
- CRIEnumerator._get_containers do not cache unavailable (404) pods if querying by filesystem (see comments)
- Change max_allowed_checkpoint_age to be a proper / configurable option
- Added note about making _K8sCache a fixed-size LRU cache

### CODE REVIEW

**Review by alesnovak-s1 — APPROVED** [review]

**Comment by alesnovak-s1** [comment]
Some of the kubernetes integration test are failing, it seems it's related to short-lived pods.
You can ingore the linux packages tests, there's a know issue, which will be handled soon.

**Review by alesnovak-s1 — CHANGES_REQUESTED** [review]
Some of the kubernetes integration test are failing, it seems it's related to short-lived pods.
You can ingore the linux packages tests, there's a know issue, which will be handled soon.

**Comment by jmakar-s1** [comment]
> Some of the kubernetes integration test are failing, it seems it's related to short-lived pods. You can ingore the linux packages tests, there's a know issue, which will be handled soon.

Ok, unit test is now fixed, ready for another look

**Comment by alesnovak-s1** [comment]
> > Some of the kubernetes integration test are failing, it seems it's related to short-lived pods. You can ingore the linux packages tests, there's a know issue, which will be handled soon.
> 
> Ok, unit test is now fixed, ready for another look

Thank you.
Can you take a look at the k8s integration tests as well?

i.e. https://github.com/scalyr/scalyr-agent-2/actions/runs/9175077249/job/25228143728

**Comment by jmakar-s1** [comment]
> Thank you. Can you take a look at the k8s integration tests as well?
> 
> i.e. https://github.com/scalyr/scalyr-agent-2/actions/runs/9175077249/job/25228143728

Fixed now & ready for another look

File: .github/workflows/reusable-agent-build-container-images.yml [hunk 1]
```diff
@@ -579,17 +579,6 @@ jobs:
           ./scripts/cicd/scalyr-query.sh '$serverHost="'${SCALYR_AGENT_POD_NAME}'" $logfile="/var/log/scalyr-agent-2/kubernetes_events.log" $parser="k8sEvents" k8s-kind="CronJob" involvedObjectKind="CronJob" involvedObjectName="hello" watchEventType="ADDED" reason="SawCompletedJob"'
           
           ./scripts/cicd/scalyr-query.sh '$serverHost="'${SCALYR_AGENT_POD_NAME}'" "Hello from the Kubernetes cluster"'
-
```
Comment (alesnovak-s1) [inline .github/workflows/reusable-agent-build-container-images.yml:582]: I believe that we should still have an integration test for short lived pods and including those pods based on SCALYR_K8S_INCLUDE_ALL_CONTAINERS.

**Review by alesnovak-s1 — CHANGES_REQUESTED** [review]

File: .github/workflows/reusable-agent-build-container-images.yml (hunk 1 above)
Comment (jmakar-s1) [reply .github/workflows/reusable-agent-build-container-images.yml]: Ok, ready for another look

**Review by alesnovak-s1 — APPROVED** [review]

**Pull request merged**